test/unit/utils_test.js

```diff
@@ -68,6 +68,1210 @@ describe('Utils', function() {
+          clientSignersPubKey,
+        ),
+      ).to.eql(clientSignersPubKey);
+    });
+
+    it("successfully validates server and multiple client signers, in reverse order", function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+      const clientSigners = [this.clientKP1, this.clientKP2];
+      transaction.sign(...clientSigners.reverse());
+      const clientSignersPubKey = clientSigners.map((kp) => {
+        return kp.publicKey();
+      });
+
+      const signedChallenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          signedChallenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          clientSignersPubKey,
+        ),
+      ).to.have.same.members(clientSignersPubKey.reverse());
+    });
+
+    it("successfully validates server and non-masterkey client signer", function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+      transaction.sign(this.clientKP2);
+
+      const signedChallenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          signedChallenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          [this.clientKP2.publicKey()],
+        ),
+      ).to.eql([this.clientKP2.publicKey()]);
+    });
+
+    it("successfully validates server and non-masterkey client signer, ignoring extra server signature", function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+      transaction.sign(this.clientKP2);
+
+      const signedChallenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          signedChallenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          [this.clientKP2.publicKey(), this.serverKP.publicKey()],
+        ),
+      ).to.eql([this.clientKP2.publicKey()]);
+    });
```

Comment:
The test description says we are ignoring an extra server signature, but we're ignoring an extra server signer in the list of signers, not an extra signature. An extra signature should result in a error which I see you test further down. I think the description just needs updating to say `signer` instead of `signature` at the end.